src/classes/CCON_AffiliationBackfill_TEST.cls

```diff
@@ -0,0 +1,190 @@
+            );
+            programEnrollments.add(pe);
+        }
+    }
+
+    /*********************************************************************************************************
+    * @description method responsible for creating Course Connections.
+    * @param students is a list of Contacts.
+    */
+    private static void createCourseConnections(List<Contact> students) {
+        for (Integer i=0; i<students.size(); i++) {
+            Course_Enrollment__c conn = new Course_Enrollment__c(
+                Contact__c = students[i].Id,
+                Program_Enrollment__c = programEnrollments[i].Id,
+                Course_Offering__c = offeringsList[i].Id
+            );
+            conn.put('RecordTypeId', UTIL_CustomSettingsFacade.getSettings().Student_RecType__c);
+            studentConnections.add(conn);
+        }
+    }
+
+    /*********************************************************************************************************
+    * @description testing the backfill functionality if Affiliation and Account lookup fields on existing
+    * Student Course Connection are populated if Affiliation lookup field is blank.
+    */
+    @isTest
+    public static void checkBackfillSuccess()  {
+        UTIL_CustomSettingsFacade.getSettingsForTests(
+            new Hierarchy_Settings__c(
+                Enable_Course_Connections__c = true,
+                Populate_Affil_on_Student_Course_Cxn__c = true,
+                Faculty_RecType__c = UTIL_Describe_API.getFacultyConnectionRecType(),
+                Student_RecType__c = UTIL_Describe_API.getStudentConnectionRecType()
+            )
+        );
+
+        List<Contact> studentContacts = [SELECT Id, LastName
+                                         FROM Contact
+                                         WHERE LastName LIKE 'Backfill%' LIMIT 20];
+        List<Course_Enrollment__c> studentConnections = [SELECT Id, Program_Enrollment__c, Affiliation__c, Contact__c
+                                                         FROM Course_Enrollment__c
+                                                         WHERE Affiliation__c = NULL AND Program_Enrollment__c != NULL
+														 AND Contact__c = :studentContacts];
+
+        for (Course_Enrollment__c studentCE : studentConnections) {
+            System.assertEquals(NULL, studentCE.Affiliation__c);
+        }
+
+        Test.StartTest();
+        CCON_AffiliationBackfill_BATCH batch = new CCON_AffiliationBackfill_BATCH();
+        ID ApexJobId = Database.executeBatch(batch, 200);
+        Test.stopTest();
+
+        List<Course_Enrollment__c> studentConns = [SELECT Id, Affiliation__c, Account__c
+                                                   FROM Course_Enrollment__c WHERE Id = :studentConnections];
+
+        for (Course_Enrollment__c ce : studentConns) {
+            System.assertNotEquals(NULL, ce.Affiliation__c);
+            System.assertNotEquals(NULL, ce.Account__c);
+        }
```

Comment:
NIT: something looks off here as well with the curly braces alignment...

Reply:
@lifewithryan , sorry I'm not sure what's off? Is it because I have the assert inside the for loop?

Reply:
@kdy618 it was a NIT because it looks like the brace on line 188 is directly above the brace on line 189
![Screen Shot 2020-01-30 at 12 09 05 PM](https://user-images.githubusercontent.com/334111/73477173-5aa2f200-4359-11ea-99a0-bf0546335721.png)

Reply:
@lifewithryan  ooh gotcha thank you!